airflow/ti_deps/deps/trigger_rule_dep.py

```diff
@@ -49,33 +50,46 @@ def _get_dep_statuses(self, ti, session, dep_context):
             yield self._passing_status(reason="The task had a dummy trigger rule set.")
             return
 
-        # TODO(unknown): this query becomes quite expensive with dags that have many
-        # tasks. It should be refactored to let the task report to the dag run and get the
-        # aggregates from there.
```

Comment:
Does this change actually solve the issue mentioned that this query is expensive?

Reply:
yes,  it does the query once for every dagrun instead of multiply it with tasks.